### ISSUE TITLE
[RFC] Add rule for pinning requirement versions in PHP package managers (Composer, PECL, docker-php-ext-install)

### ISSUE THREAD

**Issue by phansys** [hadolint/hadolint#937]
- [ ] This is a bug report
- [x] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

Warning about unpinned versions required by a package manager.

### Actual behavior

No warnings.

### Steps to reproduce the behavior

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
2.12.0
```

**Dockerfile** (if relevant)

```Dockerfile
(paste content of your Dockerfile)
```

**Additional environment details (OS, stack version, etc.)**

1. https://getcomposer.org/doc/03-cli.md#install-i (the accepted arguments for the `require`/`update` commands are very similar to what is used for NPM in [DL3016](https://github.com/hadolint/hadolint/blob/6c4632387a3485a9b7e1bf46c58da697178e02e2/src/Hadolint/Rule/DL3016.hs#L1)).
2. https://pecl.php.net/ 
3. https://github.com/docker-library/php/blob/master/docker-php-ext-install (this is a Docker specific helper that uses the same package registry than PECL).

<details>
<summary>PECL</summary>

```
pecl help install
pecl install [options] [channel/]<package> ...
Installs one or more PEAR packages.  You can specify a package to
install in four ways:

"Package-1.0.tgz" : installs from a local file

"http://example.com/Package-1.0.tgz" : installs from
anywhere on the net.

"package.xml" : installs the package described in
package.xml.  Useful for testing, or for wrapping a PEAR package in
another package manager such as RPM.

"Package[-version/state][.tar]" : queries your default channel's server
(pear.php.net) and downloads the newest package with
the preferred quality/state (stable).

To retrieve Package version 1.1, use "Package-1.1," to retrieve
Package state beta, use "Package-beta."  To retrieve an uncompressed
file, append .tar (make sure there is no file by the same name first)

To download a package from another channel, prefix with the channel name like
"channel/Package"

More than one package may be specified at once.  It is ok to mix these
four ways of specifying packages.
```
</details>

##### Composer:

```bash
composer require <vendor>/<package>
```

```bash
composer require <vendor>/<package>:*
```

##### PECL:

```bash
pecl install [channel/]<package>
```

##### Composer:

```bash
composer update <vendor>/<package>:^4.2
```

```bash
composer require <vendor>/<package>:^4.2
```

```bash
composer require vendor/<package>:~4.2
```

```bash
composer require <vendor>/<package>:4.5.0
```

```bash
composer require <vendor>/<package> 4.5.0
```

```bash
composer require <vendor>/<package>=4.5.0
```

```bash
composer require <vendor>/<package>:4.5@dev
```

```bash
composer require <vendor>/<package>:dev-master#2eb0c0978d290a1c45346a1955188929cb4e5db7
```

##### PECL:

```bash
pecl install [channel/]<package>-<version>
```
---

I can provide more examples, if required.
Thank you in advance.
